lib/Parse/ParseDecl.cpp

```diff
@@ -2829,6 +2828,22 @@ void Parser::ParseDeclarationSpecifiers(DeclSpec &DS,
       AttrsLastTime = true;
       continue;
 
+    case tok::kw__Checked:
+      // Checked C - it is parsed as checked array or checked function keyword
+      // checked array type, checked []
+      if (NextToken().is(tok::l_square)) {
+        if (!isCXX11AttributeSpecifier())
```

Comment:
`!getLangOpts().CPlusPlus11` was true, so the if-statement 'if (!getLangOpts().CPlusPlus11 '' !isCXX11AttributeSpecifier()) goto DoneWithDeclSpec` should just be replaced with `goto DoneWithDeclSpec`.